### PR TITLE
Remove padding without breaking interact style.

### LINKED ISSUE #7867: Default DOMWidget Padding should be 0px, and Button styling issue
@jdfreder

padding

### PR DESCRIPTION
**Pull request by jdfreder**
superset of #7868
closes #7867
without re-opening #7139

### CODE REVIEW

**Comment by jdfreder** [comment]
@SylvainCorlay

**Comment by SylvainCorlay** [comment]
OK, looking into it.

File: IPython/html/static/style/ipython.min.css [hunk 1]
```diff
@@ -1195,6 +1195,10 @@ h6:hover .anchor-link {
   font-size: 100%;
   font-style: italic;
 }
+.widget-interact div,
```
Comment (SylvainCorlay) [inline IPython/html/static/style/ipython.min.css]: `.widget-interact>div ?`

File: IPython/html/static/style/style.min.css [hunk 2]
```diff
@@ -9978,6 +9978,10 @@ h6:hover .anchor-link {
   font-size: 100%;
   font-style: italic;
 }
+.widget-interact div,
```
Comment (SylvainCorlay) [inline IPython/html/static/style/style.min.css]: idm

File: IPython/html/static/widgets/less/widgets.less [hunk 3]
```diff
@@ -1,6 +1,13 @@
 @widget-width: 350px;
 @widget-width-short: 150px;
 
+// Pad interact widgets by default.
+.widget-interact {
```
Comment (SylvainCorlay) [inline IPython/html/static/widgets/less/widgets.less]: idem and for input as well?

File: IPython/html/static/style/ipython.min.css (hunk 1 above)
Comment (jdfreder) [inline IPython/html/static/style/ipython.min.css]: Ah yes, thanks for noticing that.

**Comment by jdfreder** [comment]
@SylvainCorlay should be fixed.

**Comment by jdfreder** [comment]
@minrk @takluyver I don't know if this is still an appropriate for 3.0 fix (it's still mostly styling).  I'll leave it up to you two to decide, marked as 3.1 otherwise.

**Comment by SylvainCorlay** [comment]
I like this solution. We have alternate layout classes (I don't use core IPython's flexbox), and your change does not have side effects for me.

**Comment by takluyver** [comment]
I'd say leave it for 3.1, since CSS fixes seem to have a high probability of introducing new CSS bugs*. But Min understands this stuff better than I do, so I'll let him make the call.

* CSS critical mass: the amount of CSS wherein each bugfix occasions, on average, >1 new problem, leading to an exponential increase in bugs.

250px-castle_romeo

**Comment by SylvainCorlay** [comment]
@takluyver 
This looks *really* broken: 
padding

**Comment by takluyver** [comment]
Perhaps, but the way that the initially proposed fix made all `@interact` widgets look really broken is a good illustration of what I mean about each fix introducing other bugs. I know that one is now taken care of, but we don't have long to find out what else this fix might break.

I consider manually creating HBox containers to be a fairly advanced feature, so I don't think it's a big deal if it looks off when 3.0 goes out. I presume you can work around it by setting attributes on the widgets inside it.

**Comment by jdfreder** [comment]
> This looks really broken:

It's okay, bugs exist in software.  At least we are aware of them! :)  It does have an easy workaround too:

```python
HBox([Button(padding=0), Text(padding=0)])
```

**Comment by minrk** [comment]
Since ~nothing about widgets is going to be anywhere near remotely stable for the duration of 3.x, I'm fine with this for 3.0. 3.1 won't be so long from now, if it introduces something funky.

**Comment by jasongrout** [comment]
@takluyver: so you have a problem with the CSS styling, so you say to yourself, "I know, I'll just add another line of CSS".  Now you have two problems.

**Pull request merged**